### PR TITLE
Removed imports from subpackages from `plasmapy/utils/__init__.py`

### LINKED ISSUE #921: Discontinue importing functions, classes, & exceptions directly into plasmapy.utils
Right now `plasmapy/utils/__init__.py` includes a bunch of direct imports of exceptions, etc. from, e.g., `exceptions.py`.  Right now we're able to do things like this:
```Python
>>> from plasmapy.utils import PhysicsError, call_string
```
Because the contents of `plasmapy.utils` are highly varied, I think it would ultimately be cleaner if we didn't import functions, classes, & exceptions directly into `plasmapy.utils` but rather we imported the modules and sub-sub-packages, so that we would have to do things like
```Python
>>> from plasmapy.utils.exceptions import PhysicsError
>>> from plasmapy.utils.formatting import call_string  # after #920 is merged
```
This would prevent namespace pollution in `plasmapy.utils` (which is not a problem yet) and keep things a bit better organized.

### PR DESCRIPTION
**Pull request by tomasz-adam-skrzypczak**
## Description

I've deleted imports from sub-packages from pyplasma.utils, also I've adjusted imports in code

## Motivation and context

Resolving issue #921, this would prevent from namespace pollution

## Related issues

Closes #921

### CODE REVIEW

**Review by namurphy — CHANGES_REQUESTED** [review]
Thank you for doing this!  This is quite helpful.  I have a few minor suggestions, and then we can get this merged pretty quickly.

Could you also add a changelog entry at `changelog/2403.breaking.rst` that mentions that the functions and classes in `plasmapy.utils` need to be imported directly from the specific module in `plasmapy.utils`?

Thank you again!

File: CITATION.cff [hunk 1]
```diff
@@ -588,3 +588,6 @@ authors:
   family-names: Ahamed
   affiliation: OpenRefactory Inc.
   alias: fazledyn-or
+
+- given-names: Tomasz Adam
+  family-names: Skrzypczak
```
Comment (namurphy) [inline CITATION.cff:593]:
```suggestion
  family-names: Skrzypczak
  alias: tomasz-adam-skrzypczak
```

File: plasmapy/utils/__init__.py [hunk 2]
```diff
@@ -2,36 +2,3 @@
 Package of functions and classes used to develop clean, readable, and informative
 code.
 """
```
Comment (namurphy) [inline plasmapy/utils/__init__.py:4]:
```suggestion
"""
    __all__ = [
        "code_repr",
        "datatype_factory_base",
        "decorators",
        "exceptions",
        "roman",
    ]

from plasmapy.utils import (
    code_repr,
    datatype_factory_base,
    decorators,
    exceptions,
    roman,
)
```
I'm currently inclined to keep these imports here and in `__all__` to help with findability of the modules, and also to make sure everything still shows up in the documentation.  Thank you again!

**Comment by ejohnson-96** [comment]
Looks good, with the addition of @namurphy's comment on the citation.cff file and a changelog this should be good to go 👍

File: CITATION.cff (hunk 1 above)
Comment (tomasz-adam-skrzypczak) [reply CITATION.cff:593]: Added

File: plasmapy/utils/__init__.py (hunk 2 above)
Comment (tomasz-adam-skrzypczak) [reply plasmapy/utils/__init__.py:4]: Reverted!

File: changelog/2403.breaking.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+Removed sub-packages' imports from ``plasmapy.utils``, all classes and functions from the sub-packages have to be imported directly
```
Comment (namurphy) [inline changelog/2403.breaking.rst:1]:
```suggestion
Removed imports from subpackages in `plasmapy.utils`. All classes and functions from the subpackages now must be imported directly.
```

**Review by namurphy — APPROVED** [review]
Thank you again so much!  I made a minor edit to the changelog entry, and will set it to automerge once tests pass again.

**Pull request merged**